include/SDL3/SDL_time.h

```diff
@@ -0,0 +1,244 @@
+
+/**
+ *
+ * Converts a calendar time to seconds and nanoseconds since the epoch.
+ * This function ignores the day_of_week member of the SDL_DateTime struct, so it may remain unset.
+ *
+ * \param dt the source SDL_DateTime
+ * \param ticks the resulting SDL_Time
+ * \returns 0 on success or -1 on error; call SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC int SDLCALL SDL_DateTimeToTime(const SDL_DateTime *dt, SDL_Time *ticks);
+
+/**
+ *
+ * Given an SDL_DateTime struct, check if the members are within valid ranges to represent a real date
+ * and time.
+ *
+ * \param dt the SDL_DateTime
+ * \returns SDL_TRUE if all struct members are within valid ranges, otherwise SDL_FALSE; call SDL_GetError()
+ *          for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC SDL_bool SDLCALL SDL_DateTimeIsValid(const SDL_DateTime *dt);
+
+/* Converts an SDL time into a Windows FILETIME (100-nanosecond intervals since January 1, 1601).
+ *
+ * This function fills in the two 32-bit values of the FILETIME structure.
+ *
+ * \param ticks the time to convert
+ * \param dwLowDateTime a pointer filled in with the low portion of the Windows FILETIME value
+ * \param dwHighDateTime a pointer filled in with the high portion of the Windows FILETIME value
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern DECLSPEC void SDLCALL SDL_TimeToWindows(SDL_Time ticks, Uint32 *dwLowDateTime, Uint32 *dwHighDateTime);
+
+/* Converts a Windows FILETIME (100-nanosecond intervals since January 1, 1601) to an SDL time
+ *
+ * This function takes the two 32-bit values of the FILETIME structure as parameters.
+ *
+ * \param dwLowDateTime the low portion of the Windows FILETIME value
+ * \param dwHighDateTime the high portion of the Windows FILETIME value
+ * \returns the converted SDL time
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern DECLSPEC SDL_Time SDLCALL SDL_TimeFromWindows(Uint32 dwLowDateTime, Uint32 dwHighDateTime);
+
+/**
+ *
+ * Get the number of days in a month for a given year.
+ *
+ * \param year the year
+ * \param month the month [1-12]
+ * \returns the number of days in the requested month, otherwise -1; call SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC int SDLCALL SDL_GetDaysInMonth(int year, int month);
+
+/**
+ *
+ * Get the day of year for a calendar date.
+ *
+ * \param year the year component of the date
+ * \param month the month component of the date
+ * \param day the day component of the date
+ * \returns the day of year [0-365] if the date is valid, otherwise -1; call SDL_GetError()
+ *          for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC int SDLCALL SDL_GetDayOfYear(int year, int month, int day);
```

Comment:
What's the use case for this?

Reply:
It's useful for date arithmetic, such as if someone wants to determine the current week of the year. It's not exactly trivial to do and there's already optimized code in an internal function that can do this very efficiently, so it made sense to expose it.

Reply:
It's also useful for operating with the C standard library, as if you want to populate a struct tm, it needs the day of year to be complete (certain strftime format characters require it).